tests/framework/artifacts.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Define classes for interacting with CI artifacts in s3."""
+
+import os
+import platform
+import re
+
+from shutil import copyfile
+from typing import List
+from os import path
+from pathlib import Path
+
+import boto3
+import botocore.client
+
+
+MICROVM_CONFIG_EXTENSION = ".json"
+MICROVM_KERNEL_EXTENSION = ".bin"
+MICROVM_DISK_EXTENSION = ".ext4"
+
+
+class Artifact:
+    """A generic artifact manipulation class."""
+
+    bucket = None
+    key = None
+    local_folder = None
```

Comment:
If `local_folder` is left initialized to `None`, an immediate call to `local_dir()` after instantiation would yield `None` as path.

It would be good to move the variables inside the `__init__` function and perhaps check for uninitialized values for `local_folder`. If `local_dir()` and `local_path()` are not designed to be public, the functions should also be marked as private.

Reply:
Done